server/svix-server/src/v1/endpoints/application.rs

```diff
@@ -293,7 +294,7 @@ async fn update_application(
     Path(ApplicationPath { app_id }): Path<ApplicationPath>,
     permissions::Organization { org_id }: permissions::Organization,
     ValidatedJson(data): ValidatedJson<ApplicationIn>,
-) -> Result<(StatusCode, Json<ApplicationOut>)> {
+) -> Result<JsonStatusUpsert<200, 201, ApplicationOut>> {
```

Comment:
```suggestion
) -> Result<JsonStatusUpsert<ApplicationOut>> {
```
Nitpick

It already feels like a special case, and we'll have this exact behavior everywhere anyway, so may as well just assume the status codes? Seeing both here, especially since they are in reverse order (create is second, update is first, though I guess order doesn't matter here).

Reply:
Yeah, on second thought I agree, `upsert` is a special case anyway. Hardcoded this one to 200-UPDATE/201-CREATE.